adapters/repos/db/shard_replication.go

```diff
@@ -117,15 +132,12 @@ func (s *Shard) prepareDeleteObject(ctx context.Context, requestID string, uuid
 }
 
 func (s *Shard) preparePutObjects(ctx context.Context, requestID string, objects []*storobj.Object) replica.SimpleResponse {
-	if s.isReadOnly() {
-		return replica.SimpleResponse{Errors: []string{storagestate.ErrStatusReadOnly.Error()}}
-	}
 	task := func(ctx context.Context) interface{} {
 		rawErrs := s.putBatch(ctx, objects)
-		resp := replica.SimpleResponse{Errors: make([]string, len(rawErrs))}
+		resp := replica.SimpleResponse{Errors: make([]replica.Error, len(rawErrs))}
 		for i, err := range rawErrs {
 			if err != nil {
-				resp.Errors[i] = err.Error()
+				resp.Errors[i] = replica.Error{Code: replica.StatusConflict, Msg: err.Error()}
```

Comment:
Can we be sure the error scenario will always be a conflict here?

Reply:
Right now everything which prevents the commit()  from being executed is marked as conflict.